Samples/PushNotifications/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,195 @@
+}
+
+int main()
+{
+    // Initialize dynamic dependencies so we can consume the Windows App SDK APIs in the Windows App SDK framework package from this unpackaged app. 
+
+    // Major.Minor version, MinVersion=0 to find any framework package for this major.minor version
+    const UINT32 c_Version_MajorMinor{ 0x00040001 };
+    const PACKAGE_VERSION minVersion{};
+    RETURN_IF_FAILED(MddBootstrapInitialize(c_Version_MajorMinor, nullptr, minVersion));
+
+    PushNotificationActivationInfo info(PushNotificationRegistrationActivators::ProtocolActivator);
+    PushNotificationManager::RegisterActivator(info);
+
+   
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+    switch (kind)
+    {
+
+    case ExtendedActivationKind::Launch:
+    {
+
+        // request a channel
+        PushNotificationChannel channel = RequestChannel();
+
+        // register the chaneel
+        if (channel)
+        {
+            auto channelUri = channel.Uri();
+            DateTimeFormatter formater = DateTimeFormatter(L"on {month.abbreviated} {day.integer(1)}, {year.full} at {hour.integer(1)}:{minute.integer(2)}:{second.integer(2)}");
+
+            std::cout << "Channel Uri: " << winrt::to_string(channelUri.ToString()) << std::endl << std::endl;
+            std::wcout << L"Channel Uri will expire " << formater.Format(channel.ExpirationTime()).c_str() << std::endl;
+
+            // Register Push Event for Foreground
+            winrt::event_token token = channel.PushReceived([](const auto&, PushNotificationReceivedEventArgs const& args)
+                {
+                    auto payload = args.Payload();
+
+                    // Do stuff to process the raw payload
+                    std::string payloadString(payload.begin(), payload.end());
+                    std::cout << "Push notification content received from FOREGROUND: " << payloadString << std::endl << std::endl;
+                    args.Handled(true);
+                });
+        }
+
+        std::cout << "Press 'Enter' at any time to exit App." << std::endl;
+        std::cin.ignore();
+    }
+    break;
+
+    case ExtendedActivationKind::Push:
+    {
+        PushNotificationReceivedEventArgs pushArgs = args.Data().as<PushNotificationReceivedEventArgs>();
+
+        // Call GetDeferral to ensure that code runs in low power
+        auto deferral = pushArgs.GetDeferral();
+
+        auto payload = pushArgs.Payload();
+
+        // Do stuff to process the raw payload
+        std::string payloadString(payload.begin(), payload.end());
+        std::cout << "Push notification content received from BACKGROUND: " << payloadString.c_str() << std::endl;
+        std::cout << "Press 'Enter' to exit the App." << std::endl;
+
+        // Call Complete on the deferral when finished processing the payload.
+        // This removes the override that kept the app running even when the system was in a low power mode.
+        //deferral.Complete();
```

Comment:
Please uncomment this.

Reply:
Done, it was commented out due ot a compile issue. (it was amissing header)